src/server/plugins/dependency.ts

```diff
@@ -0,0 +1,9 @@
+import fp from 'fastify-plugin';
+import { FastifyPluginAsync } from 'fastify';
+import { AuthHandler } from '../../auth/handler';
+
+const authFuncs = new AuthHandler();
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const dependencyPlugin: FastifyPluginAsync = fp(async (server, _) => {
+  server.decorate('service', authFuncs);
```

Comment:
Its not handler its service, and pass in as object as other service needs to be declared